aten/src/ATen/native/native_functions.yaml

```diff
@@ -6472,6 +6472,34 @@
     SparseCPU, SparseCUDA: norm_sparse
   autogen: native_norm.ScalarOpt_dim_dtype_out
 
+# return values are: output, save_mean, save_var, reserve
+# reserve tensor is only used in cudnn
+- func: batch_norm_with_update(Tensor input, Tensor? weight, Tensor? bias, Tensor(a!) running_mean, Tensor(b!) running_var, float momentum, float eps, bool cudnn_enabled) -> (Tensor, Tensor, Tensor, Tensor)
+  dispatch:
+    CPU: batch_norm_with_update_cpu
+    CUDA: batch_norm_with_update_cuda
+    MPS: batch_norm_with_update_mps
+    MkldnnCPU: batch_norm_with_update_mkldnn
+  autogen: batch_norm_with_update_functional
+
+- func: batch_norm_no_update(Tensor input, Tensor? weight, Tensor? bias, Tensor? running_mean, Tensor? running_var, float momentum, float eps, bool cudnn_enabled) -> (Tensor, Tensor, Tensor, Tensor)
```

Comment:
I didn't notice the last argument the first time around: why do we need a `cudnn_enabled` bool? Shouldn't this op being using your `_select_batch_norm_backend` logic to decide whether or not to dispatch to cudnn?

Reply:
This arg is actually passed in from `aten.batch_norm` [here](https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/Normalization.cpp#L643). Today we check if `cudnn_enabled` is True is one of the criteria for [actually using cudnn](https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/Normalization.cpp#L542) (and miopen). In my new op I'm trying to preserve this behavior, but I agree it's very confusing. Do you think I should just ignore the `cudnn_enabled` passed in from `aten.batch_norm` instead?

Reply:
Oh and it looks like it's first plumbed from `torch.backends.cudnn.enabled` here in python: https://github.com/pytorch/pytorch/blob/main/torch/nn/functional.py#L2510.

So...

Option (1): preserve the behavior (generic batch_norm op gets a `cudnn_enabled` bool, which to be fair is also the case for `aten.batch_norm` today)

Option (2): don't do this, and figure out how to preserve the current behavior? One option is from C++ we read off the value of `torch.backends.cudnn.enabled` - although doing this via pybind has a perf cost, which might not be ok for eager.

I would tentatively say the current state is fine? (@albanD in case you have opinions)

Reply:
Since we're doing the hard work of cleaning this up, I think we should do it completely.
We already set the state of the cudnn flag in c++ via `userEnabledCuDNN()` so you can just read that within the backend decision logic.

Reply:
@albanD and to confirm, do you also want to kill that flag from `aten.batch_norm` too while we're at it? (risks: `aten.batch_norm` is a CompositeImplicit op, so it's only BC breaking for anyone who serializes "pre-dispatch" ATen, which is basically just.... torch.jit.trace, I think?)

Reply:
Ok, I removed the `cudnn_enabled` arg from the new ops. For the existing `aten.batch_norm` op I think it's better to remove the arg in a future PR if we decide to do it. Added a TODO for now to track